workflow/lib/shared/rule_utils.py

```diff
@@ -0,0 +1,93 @@
+"""Helper functions for using Snakemake rules for use with Brieflow."""
+
+
+def filter_outputs_by_cycle_index(outputs, index):
+    """Filter and select outputs by cycle index from sorted available cycles.
+    
+    Args:
+        outputs (list): List of output paths
+        index (int): Index to select from sorted cycles (-1 for last)
+        
+    Returns:
+        str: Selected path or empty string if not found
+    """  
```

Comment:
It looks like you have to sort your cycle paths to make sure you get the right one when indexing? If I have this correct then a function for this filtering does make sense, thanks for letting me take another look!